test/configuration_test.rb

```diff
@@ -605,25 +605,47 @@ class FileSystem
     end
   end
 
-  describe '#default_format' do
+  describe '#default_request_format' do
+    describe "when not previously set" do
+      it 'returns :html' do
+        @configuration.default_request_format.must_equal :html
+      end
+    end
+
+    describe "when set" do
+      before do
+        @configuration.default_request_format :json
+      end
+
+      it 'returns the value' do
+        @configuration.default_request_format.must_equal :json
+      end
+    end
+
+    it 'raises an error if the given format cannot be coerced into symbol' do
+      -> { @configuration.default_request_format(23) }.must_raise TypeError
+    end
+  end
+
+  describe '#default_response_format' do
     describe "when not previously set" do
       it 'returns nil' do
-        @configuration.default_format.must_equal :html
+        @configuration.default_response_format.must_equal nil
```

Comment:
@jodosha I see your point. But bare in mind that if this value is nil, the framework will guess the content type by looking at the request content type.  (https://github.com/lotus/controller/blob/master/lib/lotus/action/mime.rb#L155)

The issue with setting `:html` to be the default there is that the framework will then never try to detect the request type because default_response_format will always have a value.

In other words, this logic hasn't changed. The only difference is that if `default_response_format` is set, it will use that. Otherwise it will use the same logic as before.

On the other hand, if you wish to change the logic that is currently in place and have the default_response_format always be `:html` by default, thats fine too. As long as we understand that slight change in semantics :)

please me know your thoughts